metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfiguration.scala

```diff
@@ -0,0 +1,70 @@
+package scala.meta.internal.metals
+
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+
+import scala.meta.internal.bsp.BspServers
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.ClasspathSearch
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.MtagsResolver
+import scala.meta.internal.metals.ProgressTicks
+import scala.meta.internal.metals.Time
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.io.AbsolutePath
+
+/**
+ * Container class for metals language server and service configuration.
+ *   The configuration is mostly used to allow for custom behaviour in tests.
+ *   All the defaults are the instances used on the production.
+ */
+abstract class MetalsServerConfiguration {
+  val charset: Charset = StandardCharsets.UTF_8
+
+  /** In-memory text contents for unsaved files. */
+  val buffers: Buffers = Buffers()
+
+  /** If system.out should be redirected into a file. */
+  val redirectSystemOut: Boolean = true
+
+  /** A "clock" instance for getting the current time. */
+  val time: Time = Time.system
+
+  /** Configuration parameters for the Metals language server. */
+  val initialServerConfig: MetalsServerConfig = MetalsServerConfig.default
```

Comment:
I didn't think about one thing - now we will have `MetalsServerConfig` and `MetalsServerConfiguration`. Maybe it makes sense to use `MetalsServerInputs` because it is bag of inputs? 🤔 
@tgodzik what's your opinion?

Reply:
Renamed to `MetalsServerInputs` for now.